gordon/record_checker.py

```diff
@@ -0,0 +1,112 @@
+import asyncio
+import logging
+import time
+
+from async_dns import types
+from async_dns.resolver import ProxyResolver
+
+
+class RecordChecker(object):
+    QUERYING_THRESHOLD = 60
+
+    def __init__(self, dns_ip):
+        """Setup RecordChecker object
+
+        Args:
+            dns_ip: DNS ip server to query
```

Comment:
I think this could be better named to something like `dns_server_list` or something if we wanted to support querying of multiple upstreams.